### ISSUE TITLE
status and diagnostic information could not be printed for remote-ssh coinnection

### ISSUE THREAD

**Issue by hsinyinfu** [microsoft/vscode#150935]
Does this issue occur when all extensions are disabled?: Yes (only enable "Remote-SSH" and "Remote-SSH: Editing Configuration Files" extensions

- VS Code Version: 1.67.2
- Local OS Version: Windows 10 Enterprise 20H2 19042.1083
- Remote OS Version: Ubuntu 18.04.5 LTS
- Remote Extension/Connection Type: SSH

Steps to Reproduce:

1. Remote-ssh to a remote Ubuntu server
2. Add two folders on the remote Linux server into an untitled workspace
3. Run `code -s` in the Windows CMD to get status of the local and remote VS Code processes
4. The "Connecting to 'SSH: linux_terminal_server' could not be established stdout maxBuffer length exceeded" is showed in the bottom of the status report.
(The status of local VS Code processes report seems to work fine. All the hardware, local processes, and local workspace information is printed correctly.)

**Comment by hsinyinfu** [comment]
Yes, I could successfully connect to my remote Linux server with remote-ssh and everything works fine.
The only problem is that the CLI command `code -s` and the "Process Explorer" in VS Code could not get the remote Linux status (e.g., it should get information about processes and some hardware information), but only show the information about local Windows system.

**Comment by Tyriar** [comment]
@aeschli I'm unfamiliar with this, you seem like the right owner?

![image](https://user-images.githubusercontent.com/2193314/171655743-111ab18e-6c4b-4943-9d9c-7c50d09024f4.png)

![image](https://user-images.githubusercontent.com/2193314/171655624-7077cb74-2d11-44a1-bd8b-f7793e70b260.png)

**Comment by aeschli** [comment]
Rachel implemented the remote (and local) status support. I helped her wiring the remote cli to the `getSystemStatus` call which is implemented on main.
https://github.com/Microsoft/vscode/blob/main/src/vs/platform/issue/electron-main/issueMainService.ts#L353

I think that one then again reaches out to the remote to get information from the remote.

I don't know what information is expected to be shown. On WSL I see the Windows processes, but from WSL I just see a summary of basic info (OS Version, CPIs, Memory)

**Comment by hsinyinfu** [comment]
Currently, sometimes if it works correctly, it shows the process stack on the remote server (the child processes spawn by the remote-ssh server entry program got listed as long as some basic hardware information).
I think that information helps a lot for users (like me) to write some CI or automatical job to periodically monitor the status of the remote processes spawn by VS Code, remote hardware usage IMHO.

**Comment by legomushroom** [comment]
Revalidated this today on stable 1.95.3 and Ubuntu 24, the issue still exists 🐛 The error message had improved though - I get `Connection to '{HOSTNAME}' could not be established` now.
